php/class-assets.php

```diff
@@ -1020,7 +1020,8 @@ public function get_asset( $post_id, $type = 'object' ) {
 		global $wpdb;
 
 		$wpdb->cld_table = Utils::get_relationship_table();
-		$prepare         = $wpdb->prepare( "SELECT * FROM $wpdb->cld_table WHERE post_id = %d;", (int) $post_id );
+		$media_context   = Utils::get_media_context( $post_id );
+		$prepare         = $wpdb->prepare( "SELECT * FROM $wpdb->cld_table WHERE post_id = %d AND media_context = %s;", (int) $post_id, $media_context );
```

Comment:
Should we add media_context to the query in https://github.com/cloudinary/cloudinary_wordpress/blob/d2bfec53c3144f02f20e80fa75c118db5c635786/php/sync/class-sync-queue.php#L325 ?

Reply:
That's a good question @yuval-cloudinary.

The single purpose of that query you're pointing out is to get the count of unsynced assets to populate the plugin dashboard. The way it is, it's still retrieving all the assets, regardless of the context. If we filter by context, we might get a partial overlook that doesn't reflect the overall state.